plugin.js

```diff
@@ -682,189 +496,423 @@
 		};
 	}
 
-	placeholdersCollection = new PlaceholdersCollection();
+	/**
+	 * @class
+	 * @constructor
+	 * @param {CKEDITOR.editor} editor
+	 */
+	function PlaceholderElementsPlugin(editor)
+	{
+		var _this = this;
 
-	CKEDITOR.plugins.add('placeholder_elements', {
-		hidpi: true,
-		icons: 'placeholderelements',
-		lang: ['en', 'de'],
-		requires: ['richcombo', 'widget'],
+		var config = editor.config.placeholder_elements;
+		var placeholdersCollection = new PlaceholdersCollection();
+		var placeholdersRegexLoose;
+		var placeholdersRegexExact;
+		var placeholdersMap = {};
+		var uiElement;
+		var menuButtonStates = {};
+		var menuButtonGroup = 'placeholderElementsButton';
+		var contentIsUpdating = false;
+		var contentNeedsUpdating = false;
+		var selectionChangeProcessor;
 
-		placeholders: placeholdersCollection,
+		/**
+		 * @type {PlaceholdersCollection}
```

Comment:
Neither it's mine. I'm too lazy to actually put this to a test right now, so since it won't hurt I guess it's OK to keep it.